### ISSUE TITLE
failed to compile on Julia v1.0

### ISSUE THREAD

**Issue by yangcht** [JuliaPhysics/Measurements.jl#18]
when using Measurements in Julia 1.0, it fails 

```
julia> using Measurements
[ Info: Precompiling Measurements [eff96d63-e80a-5855-80a2-b1b0885c5ab7]
ERROR: LoadError: LoadError: UndefVarError: Associative not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] include at ./boot.jl:317 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1038
 [4] include at ./sysimg.jl:29 [inlined]
 [5] include(::String) at /Users/yangcht/.julia/packages/Measurements/pcvQy/src/Measurements.jl:22
 [6] top-level scope at none:0
 [7] include at ./boot.jl:317 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1038
 [9] include(::Module, ::String) at ./sysimg.jl:29
 [10] top-level scope at none:2
 [11] eval at ./boot.jl:319 [inlined]
 [12] eval(::Expr) at ./client.jl:389
 [13] top-level scope at ./none:3
in expression starting at /Users/yangcht/.julia/packages/Measurements/pcvQy/src/derivatives-type.jl:22
in expression starting at /Users/yangcht/.julia/packages/Measurements/pcvQy/src/Measurements.jl:32
ERROR: Failed to precompile Measurements [eff96d63-e80a-5855-80a2-b1b0885c5ab7] to /Users/yangcht/.julia/compiled/v1.0/Measurements/8B0my.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834

julia>
```

How should it be solved? Many thanks!!

**Comment by giordano** [comment]
Try switching to master branch, it should support julia 0.7 and thus julia 1.0

**Comment by yangcht** [comment]
Hi Mosè, I tried to switch to the master branch by

`(v1.0) pkg> add Measurements#master`

But I got a similar error when I compile it for use:
```
julia> using Measurements
[ Info: Precompiling Measurements [eff96d63-e80a-5855-80a2-b1b0885c5ab7]
ERROR: LoadError: LoadError: UndefVarError: start not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include at ./sysimg.jl:29 [inlined]
 [6] include(::String) at /Users/yangcht/.julia/packages/Measurements/oelLI/src/Measurements.jl:22
 [7] top-level scope at none:0
 [8] include at ./boot.jl:317 [inlined]
 [9] include_relative(::Module, ::String) at ./loading.jl:1038
 [10] include(::Module, ::String) at ./sysimg.jl:29
 [11] top-level scope at none:2
 [12] eval at ./boot.jl:319 [inlined]
 [13] eval(::Expr) at ./client.jl:389
 [14] top-level scope at ./none:3
in expression starting at /Users/yangcht/.julia/packages/Measurements/oelLI/src/derivatives-type.jl:66
in expression starting at /Users/yangcht/.julia/packages/Measurements/oelLI/src/Measurements.jl:32
ERROR: Failed to precompile Measurements [eff96d63-e80a-5855-80a2-b1b0885c5ab7] to /Users/yangcht/.julia/compiled/v1.0/Measurements/8B0my.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834

```

**Comment by giordano** [comment]
For the record, this package passed CI without warnings about 20 days before release of julia 1.0: https://travis-ci.org/JuliaPhysics/Measurements.jl/jobs/406597690  thus either there have been late breaking changes or deprecation warnings were missing.

**Comment by yangcht** [comment]
Could this be an issue related to the dependent packages?

**Comment by giordano** [comment]
No, from a cursory look at the code I think it's an issue in this package, but until a few days ago it worked without warnings

**Comment by giordano** [comment]
Fixed in #19
